tests/test_callbacks.py

```diff
@@ -203,3 +203,44 @@ def test_eval_friendly_error():
     with pytest.warns(Warning):
         with pytest.raises(AssertionError):
             model.learn(100, callback=eval_callback)
+
+
+def test_checkpoint_replay_buffer(tmp_path):
+    # tests if the replay buffer is saved with every checkpoint
+
+    n_bits = 2
+    env = BitFlippingEnv(n_bits=n_bits)
+
+    checkpoint_dir = tmp_path / "checkpoints"
+    checkpoint_callback = CheckpointCallback(save_freq=500, save_path=checkpoint_dir, save_replay_buffer=True)
+
+    model = DQN(
+        "MultiInputPolicy",
+        env,
+        replay_buffer_class=HerReplayBuffer,
+        learning_starts=100,
+        seed=0,
+        replay_buffer_kwargs=dict(max_episode_length=n_bits),
+    )
+    model.learn(500, callback=checkpoint_callback)
+
+    assert os.path.exists(checkpoint_dir / "rl_model_replay_buffer_500_steps.pkl")
+
+
+def test_checkpoint_vecnormalize(tmp_path):
```

Comment:
you can probably combine both tests? or was there a reason to use `HerReplayBuffer` ?
and you should set `buffer_size` to a small value to avoid using too much disk space.

Reply:
I thought I would test the functionality separately. But I can merge them too.